fem/gridfunc.hpp

```diff
@@ -916,6 +920,48 @@ double ZZErrorEstimator(BilinearFormIntegrator &blfi,
                         int with_subdomains = 1,
                         bool with_coeff = false);
 
+/// Defines the global tensor product polynomial space used by NewZZErorrEstimator
+/**
+ *  See BoundingBox(...) for a description of @a angle and @a midpoint
+ */
+void TensorProductLegendre(int dim,                      // input
+                           int order,                    // input
+                           const Vector &x_in,           // input
+                           const Vector &xmax,           // input
+                           const Vector &xmin,           // input
+                           Vector &poly,                 // output
+                           double angle=0.0,             // input (optional)
+                           const Vector *midpoint=NULL); // input (optional)
+
+/// Defines the bounding box for the face patches used by NewZZErorrEstimator
+/**
+ *  By default, BoundingBox(...) computes the parameters of a minimal bounding box
+ *  for the given @a face_patch that is aligned with the physical (i.e. global)
+ *  Cartesian axes. This means that the size of the bounding box with depend on the
+ *  orientation of the patch. It is better to contruct an orientation-independent box.
+ *  This is implemented for patches made up of quadrilaterals. The parameters @a angle
+ *  and @a midpoint encode this additional geometric information.
+ *
+ *      @a angle     : the angle the patch face makes with the x-axis
+ *      @a midpoint  : the midpoint of the face
+ */
+void BoundingBox(Array<int> face_patch,    // input
+                 FiniteElementSpace *ufes, // input
+                 int order,                // input
+                 Vector &xmin,             // output
+                 Vector &xmax,             // output
+                 double &angle,            // output
+                 Vector &midpoint,         // output
+                 int iface=-1);            // input (optional)
+
+/// A ``true'' ZZ error estimator that uses face-based patches for flux reconstruction
+double NewZZErrorEstimator(BilinearFormIntegrator &blfi,         // input
```

Comment:
Also, I think we need a description in how the face patches are contracted in the non-confirming case, i.e only the fine faces are considered etc.